pkg/driver/credential.go

```diff
@@ -145,18 +161,27 @@ func (c *CredentialProvider) provideFromPod(ctx context.Context, req *csi.NodePu
 	}, nil
 }
 
-func (c *CredentialProvider) writeToken(volumeID string, token *Token) error {
-	// TODO: Should we make the write atomic by writing to a temporary path and renaming afterwards?
-	return os.WriteFile(c.tokenPath(volumeID), []byte(token.Token), 0400)
+func (c *CredentialProvider) writeToken(podID string, volumeID string, token *Token) error {
+	return WriteFileAtomic(c.tokenPathContainer(podID, volumeID), []byte(token.Token), serviceAccountTokenPerm)
+}
+
+func (c *CredentialProvider) tokenPathContainer(podID string, volumeID string) string {
+	return path.Join(c.containerPluginDir, c.tokenFilename(podID, volumeID))
 }
 
-func (c *CredentialProvider) tokenPath(volumeID string) string {
-	return path.Join(c.containerPluginDir, volumeID+".token")
+func (c *CredentialProvider) tokenFilename(podID string, volumeID string) string {
```

Comment:
Do we also need to include namespaces? Is it plausible that pods in different namespaces could have the same id?

Reply:
Nevermind, according to [the docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/), UUIDs are unique across a cluster